src/App.tsx

```diff
@@ -14,11 +14,95 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  setSelectedGoods = (goods: string) => {
+    this.setState(((prevstate: State) => {
+      return { selectedGoods: [...prevstate.selectedGoods, goods] };
+    }));
+  };
+
+  removeSelectedGood = (good: string) => {
+    this.setState((prevState => ({
+      selectedGoods: prevState.selectedGoods.filter(remaining => (
+        remaining !== good)),
+    }
+    )));
+  };
+
+  clearSelectedGoods = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  createTitle = (goodsArr: string[]) => {
+    const { length } = goodsArr;
+
+    switch (length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${goodsArr[0]} is selected`;
+      case 2:
+        return `${goodsArr.join(' and ')} are selected`;
+      default:
+        return `${goodsArr.slice(0, -1).join(', ')} and ${goodsArr[length - 1]} are selected`;
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="select">
+        <div>
+          <h1 className="select__content">
+            {this.createTitle(selectedGoods)}
+          </h1>
+
+          <button
+            type="button"
+            className="select__clearBtn"
+          >
+            clear
+          </button>
```

Comment:
`onClick` has not been set for the button. Clear button therefore does not work.

Reply:
fixed